sycl/unittests/scheduler/ImageApi.cpp

```diff
@@ -0,0 +1,73 @@
+//==----------------- ImageApi.cpp --- Scheduler unit tests ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "MockHandler.hpp"
+#include "SchedulerTest.hpp"
+
+#include <CL/sycl.hpp>
+
+#include <gtest/gtest.h>
+
+TEST_F(SchedulerTest, ImageApi) {
+
+  constexpr size_t Size = 256;
+  constexpr int Dimensions = 2;
+
+  std::array<sycl::float4, Size> Src;
+  std::array<sycl::float4, Size> Dest;
+  std::fill(Src.begin(), Src.end(), sycl::float4{1.0f, 2.0f, 3.0f, 4.0f});
+  std::fill(Dest.begin(), Dest.end(), sycl::float4{0.0f, 0.0f, 0.0f, 0.0});
+
+  constexpr sycl::image_channel_order ChannelOrder =
+      sycl::image_channel_order::rgba;
+  constexpr sycl::image_channel_type ChannelType =
+      sycl::image_channel_type::fp32;
+  const sycl::range<Dimensions> Range{16, 16};
+
+  sycl::image<Dimensions> SrcImg(Src.data(), ChannelOrder, ChannelType, Range);
+  sycl::image<Dimensions> DstImg(Dest.data(), ChannelOrder, ChannelType, Range);
+
+  auto CGHLambda = [&](MockHandler &CGH) {
+    auto SrcAcc =
+        SrcImg.template get_access<sycl::float4, sycl::access::mode::read>(CGH);
```

Comment:
I'm not sure this test makes sense when it doesn't use internal API.
Why can't we continue using internal API in google tests?
And if we cannot, I suggest removing the test completely.

Reply:
By "use internal API" do you mean creating accessor directly inside the handler?
I think it is a duplication of code. And if the logic inside the handler changes a bit, this test will become invalid (yet it still may pass). Do we need such a test?

Reply:
Let's remove it.